sunpy/net/jsoc/jsoc.py

```diff
@@ -34,32 +35,23 @@ class NotExportedError(Exception):
     pass
 
 
-class JSOCResponse(BaseQueryResponseTable):
-    def __init__(self, table=None, client=None):
-        """
-        table : `astropy.table.Table`
-        """
-        super().__init__(table, client)
-        self.query_args = getattr(table, 'query_args', None)
-        self.requests = getattr(table, 'requests', None)
-
-    def __getitem__(self, item):
-        ret = super().__getitem__(item)
-        ret.query_args = self.query_args
-        ret.client = self._client
-        warnings.warn("Downloading of sliced JSOC results is not supported. "
-                      "All the files present in the original response will be downloaded.",
-                      SunpyUserWarning)
-        return ret
+class JSOCResponse(QueryResponseTable):
+    query_args = TableAttribute()
+    display_keys = ['T_REC', 'TELESCOP', 'INSTRUME', 'WAVELNTH', 'CAR_ROT']
+    _original_num_rows = TableAttribute(default=None)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._original_num_rows = len(self)
 
+    # remove this method post 3.0
```

Comment:
```suggestion
    # TODO: remove this method post 3.0
```
Makes it easy to find.